gcc/rust/expand/rust-macro-builtins.cc

```diff
@@ -949,9 +949,12 @@ MacroBuiltin::sorry (Location invoc_locus, AST::MacroInvocData &invoc)
 }
 
 AST::Fragment
-MacroBuiltin::proc_macro_builtin (Location, AST::MacroInvocData &)
+MacroBuiltin::proc_macro_builtin (Location invoc_locus,
+				  AST::MacroInvocData &invoc)
 {
-  // nothing to do!
+  rust_error_at (invoc_locus, "cannot invoke derive macro: %qs",
+		 invoc.get_path ().as_string ().c_str ());
```

Comment:
I was thinking that it would produce the error when the user attempts to invoke a derive macro like [so](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=b0cdd04f02f9449e29a2c24251022366)

Reply:
Ah, good point! One small correction is that the proper invocation would be [this](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=9f2c711e50925ccf9dca1f54ad1a86a1) (so using `core::marker::Copy!()` instead of `Copy!()`, which fails because of name resolution). So it seems `rustc` does special case these invocations and your fix is perfect :) good job